### ISSUE TITLE
DOC precisions - Track an untracked file dependency as a component dependency

### ISSUE THREAD

**Issue by ThomasKientz** [teambit/bit#1233]
https://docs.bitsrc.io/docs/isolating-and-tracking-components.html#track-an-untracked-file-dependency-as-a-component-dependency

> In order to resolve the issue from the previous example, and set src/utils/noop.js as a new component dependency for hello/world you need to run the add command.
> `$ bit add src/utils/noop.js --id utils/noop`
> The result will be a new component, which is now a dependency for the hello/world component.

The doc doesn't explained if we need or not to change the import in hello-world.js :

```
// hello-world.js

import noop from '../utils/noop'; // do we need to change this with "@bit/..." ?

export default function hello(world) {
    noop();
    return `hello ${world}`;
}
```

**Comment by ThomasKientz** [comment]
In the doc "Importing Components",  ["Binding"](https://docs.bitsrc.io/docs/importing-components.html#bindings
) section :

we can read :

> Tip
> To reduce and flatten a component file system structure, you can modify your code to use absolute paths. For more information, please see Linking components.

**Comment by itaymendel** [comment]
hi @ThomasKientz , the reason the docs don't go to specifics, in this case, is that when working with local, authored components, it's not required to refactor the code.

do you think adding a similar `tip` after the example to the will explain it better?

**Comment by GiladShoham** [comment]
Yes, there is a difference if you refactor the code or not. The difference is only in how Bit structures imported components in the project you import the component to.

In case you do not refactor, this will be the result of the imported component file structure:

```bash
-component-dir
--utils
---noop.js (this will be an auto-generated binding file which points to the noop component in node_modules)
--other-dir
---hello-world.js
```

In case you refactor the code, this will be the result:

```bash
-component-dir
--hello-world.js
```
Note that since the `hello-world` component does not have a relative path to `noop.js`, Bit does not create the link file. This results in less clutter of files for an imported component.

> I guess there is a difference: If I don't refactor (keep a relative link to my local component) in the parent component (A), and if I update and push the subcomponent (the imported component B). Other projects who import A will not get the update of B, without updating A. Is that right ?
> If I do refactor, other projects only need to install latest version of B.

There will be no difference when updating imported components in both cases. Bit is capable of handling the updates of component B, even if A requires is using a relative path.

> One more thing the doc could be more explicit about the refactoring process. Do we need to run bit link in the process of refactoring ? So we refactor, we can still work locally. Is my question clear ?

Bit generates the links during the `export` process. Only then a component has a full name (including the scope part of it). So to your question, no, you do not need to manually run `bit link`. 

We are designing a solution that will allow a default target for a component. This will allow you to refactor your code from relative to an absolute path before you export the component. I don't have an estimation as to when this will be GA.
